sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -40,14 +42,20 @@
 import java.util.List;
 import java.util.Map;
 
+import static com.generic.core.http.models.ContentType.APPLICATION_OCTET_STREAM;
+import static com.generic.core.http.models.HttpHeaderName.CONTENT_TYPE;
+import static com.generic.core.http.models.ResponseBodyMode.BUFFER;
+import static com.generic.core.http.models.ResponseBodyMode.STREAM;
 import static com.generic.core.implementation.util.ServerSentEventUtil.NO_LISTENER_ERROR_MESSAGE;
 import static com.generic.core.implementation.util.ServerSentEventUtil.processTextEventStream;
 
 /**
  * HttpClient implementation using {@link HttpURLConnection} to send requests and receive responses.
  */
 class DefaultHttpClient implements HttpClient {
+    private static final BinaryData EMPTY_BODY = BinaryData.fromBytes(new byte[0]);
```

Comment:
@alzimmermsft Should we make this a constant in `BinaryData` itself? `BinaryData.EMPTY`?

Reply:
Yes, we should add a constant for an empty BinaryData. Possibly as its own subtype to prevent any oddities if the `instanceof` is being checked.